instrumentation/opentelemetry-instrumentation-system-metrics/tests/test_system_metrics.py

```diff
@@ -0,0 +1,723 @@
+    def setUp(self):
+        super().setUp()
+        self.implementation = python_implementation().lower()
+        self._patch_net_connections = mock.patch(
+            "psutil.net_connections", _mock_netconnection
+        )
+        self._patch_net_connections.start()
+
+    def tearDown(self):
+        super().tearDown()
+        self._patch_net_connections.stop()
+        SystemMetricsInstrumentor().uninstrument()
+
+    def test_system_metrics_instrument(self):
+        reader = InMemoryMetricReader()
+        meter_provider = MeterProvider(metric_readers=[reader])
+        system_metrics = SystemMetricsInstrumentor()
+        system_metrics.instrument(meter_provider=meter_provider)
+        metrics = reader.get_metrics()
+        metric_names = list({x.name for x in metrics})
+        self.assertEqual(len(metric_names), 17)
+
+        observer_names = [
+            "system.cpu.time",
+            "system.cpu.utilization",
+            "system.memory.usage",
+            "system.memory.utilization",
+            "system.swap.usage",
+            "system.swap.utilization",
+            "system.disk.io",
+            "system.disk.operations",
+            "system.disk.time",
+            "system.network.dropped_packets",
+            "system.network.packets",
+            "system.network.errors",
+            "system.network.io",
+            "system.network.connections",
+            "runtime.{}.memory".format(self.implementation),
+            "runtime.{}.cpu_time".format(self.implementation),
+            "runtime.{}.gc_count".format(self.implementation),
+        ]
+
+        for observer in metric_names:
+            self.assertIn(observer, observer_names)
+            observer_names.remove(observer)
+
+    def _assert_metrics(self, observer_name, reader, expected):
+        assertions = 0
+        for metric in reader.get_metrics():  # pylint: disable=protected-access
+            for e in expected:
+                if (
+                    metric.attributes == e.attributes
+                    and metric.name == observer_name
+                ):
+                    self.assertEqual(
+                        metric.point.value,
+                        e.value,
+                    )
+                    assertions += 1
+        self.assertEqual(len(expected), assertions)
+
+    def _test_metrics(self, observer_name, expected):
+        reader = InMemoryMetricReader()
+        meter_provider = MeterProvider(metric_readers=[reader])
+
+        system_metrics = SystemMetricsInstrumentor()
+        system_metrics.instrument(meter_provider=meter_provider)
+        self._assert_metrics(observer_name, reader, expected)
+        system_metrics.uninstrument()
+
+    # When this test case is executed, _get_system_cpu_utilization gets run
+    # too because of the controller thread which runs all observers. This patch
+    # is added here to stop a warning that would otherwise be raised.
+    # pylint: disable=unused-argument
```

Comment:
Is htis comment still accurate?

Reply:
The patch is still needed, but i've updated the comment.